flink-runtime/src/test/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinatorTest.java

```diff
@@ -3688,6 +3693,92 @@ public void testTimeoutWhileCheckpointOperatorCoordinatorNotFinishing() throws E
         }
     }
 
+    @Test
+    public void testTimeoutWhileCheckpointOperatorCoordinatorNotFinishing1() throws Exception {
+        // Warn: The case is fragile since a specific order of executing the tasks is required to
```

Comment:
I was struggling with the executing order to reproduce the issue, and am looking forward to some suggestions to consolidate the case.